File: robot/src/main/java/frc/team254/lib/trajectory/TrajectoryPoint.java
package frc.team254.lib.trajectory;

import frc.team5190.lib.geometry.interfaces.State;

public class TrajectoryPoint<S extends State<S>> {
    protected final S state_;
    protected final int index_;

    public TrajectoryPoint(final S state, int index) {
        state_ = state;
        index_ = index;
    }

    public S state() {
        return state_;
    }

    public int index() {
        return index_;
    }
}

File: robot/src/main/kotlin/frc/team5190/lib/trajectory/TrajectoryPoint.kt
package frc.team5190.lib.trajectory

import frc.team5190.lib.geometry.interfaces.State

class TrajectoryPoint<S : State<S>>(val state: S, private val index: Int) {

    fun state(): S {
        return state
    }

    fun index(): Int {
        return index
    }
}
